scripts/src/main/resources/scripts/functions

```diff
@@ -1257,21 +1306,61 @@ function doInstall() {
   fi
 
   # macos workaround
-  if [ ! -d "${target_path}/bin" ] && doIsMacOs && [ -d "${target_path}/Contents/Home/bin" ]
+  if [ ! -d "${target_path}/bin" ] && doIsMacOs && [ -d "${target_path}/Contents/" ]
   then
-    doEcho "Creating symlink as workaround for ${software} on MacOS"
-    local target_software_macos=${target_path%\/*}/macos
-    if [ ! -d "${target_software_macos}" ]
+    doDebug "Found MacOS app in ${target_path}"
+    local app_folder="-"
+    local folder
+    while IFS= read -r -d '' folder
+    do
+      folder="${folder/*\//}"
+      doDebug "Checking app sub-folder ${folder}"
+      if [ "${folder}" != "Contents" ] && [ "${folder}" != "Resources" ] && [ "${folder}" != "bin" ] && [ "${folder:0:1}" != "_" ]
+      then
+        if [ -d "${target_path}/Contents/${folder}/bin" ]
+        then
+          app_folder="${folder/*\//}"
+          doDebug "App folder with bin detected as ${app_folder}"
+          break
+        else
+          doDebug "Looking for executable in ${folder}"
+          local file
+          while IFS= read -r -d '' file
+          do
+            if [ -x "${file}" ] && [ "${file:(-4)}" != ".txt" ]
+            then
+              app_folder="${folder/*\//}"
+              doDebug "App folder with executable detected as ${app_folder}"
+              break
+            fi
+            done < <(find "${target_path}/Contents/${folder}/" -maxdepth 1 -type f -print0)
+        fi
+      fi
+    done < <(find "${target_path}/Contents" -maxdepth 1 -type d -print0)
+
+    if [ "${app_folder}" != "-" ] && [ -d "${target_path}/Contents/${app_folder}" ]
     then
-      mkdir -p "${target_software_macos}"
+      doEcho "Creating symlink as workaround for ${software} on MacOS"
+      local target_software_macos=${target_path%\/*}/macos
+      if [ ! -d "${target_software_macos}" ]
+      then
+        mkdir -p "${target_software_macos}"
+      fi
+      local target_software_software=${target_software_macos}/${dir}
+      doRunCommand "rm -rf '${target_software_software}'"
+      doRunCommand "mv '${target_path}' '${target_software_software}'"
+      doRunCommand "ln -sf 'macos/${dir}/Contents/${app_folder}' '${target_path}'"
+      doRunCommand "cp '${target_software_software}/.devon.software.version' '${target_path}'"
     fi
-    local target_software_software=${target_software_macos}/${dir}
-    doRunCommand "rm -rf '${target_software_software}'"
-    doRunCommand "mv '${target_path}' '${target_software_software}'"
-    doRunCommand "ln -sf 'macos/${dir}/Contents/Home' '${target_path}'"
-    doRunCommand "cp '${target_software_software}/.devon.software.version' '${target_path}'"
   fi
-
+  if [ "${target_path}" = "${DEVON_IDE_HOME}/software/${dir}" ]
+  then
+    doExtendPath "${target_path}"
+  fi
+  if [ -n "${TOOL_VERSION_COMMAND}" ] && [ "${TOOL_VERSION_COMMAND}" != "-" ] && ! doIsQuiet
+  then
+    doRunCommand "${TOOL_VERSION_COMMAND}" "verify installation of ${software}"
```

Comment:
Is it possible that the version of the tool will not be output if `TOOL_VERSION_COMMAND` is not defined, or is a future requirement that `TOOL_VERSION_COMMAND` should be defined in every commandlet. If so, I would ensure by a warning message or test that it is not forgotten.

Reply:
Yes, we have situations where `TOOL_VERSION_COMMAND` and then it is obviously intended that no such command is executed.
Examples are:
* we have commandlets that are not related to a "tool" at all (e.g. `build`). They should never run `doInstall` but there are other places where this variable is used so it also triggers if `version get|set|list` is supported or not.
* we have custom tool support and in such case we do not even know anything about the tool except "name" and "version" and where to download the *.tgz file that we extract. So this would be the case for the code you annotated with your review comment.

Reply:
> If so, I would ensure by a warning message or test that it is not forgotten.

If you forget this, then `devon «tool» version get|set|list` will all 3 not work.
We can easily test this and add it to our DoD.
Due to the explanation above we can not easily render a warning as this would also show up in undesired situations where nothing is actually wrong.

Reply:
DoD updated.